.github/workflows/deploy-config.yml

```diff
@@ -1,27 +1,33 @@
 name: Deploy config
 on:
-  push:
-    branches:
-      - '[0-9]+.[0-9]+'
+  - create
+  - delete
```

Comment:
Unfortunately, no. These events are global and unfiltered.

I suppose we could filter for version branches in the steps. *I think* `${ github.event.ref }` should hold the name of the branch (or tag) that was created or deleted.
Per https://docs.github.com/en/actions/learn-github-actions/contexts and https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#create

Maybe it would also be a viable alternative to not automate this in this repo, but integrate the steps with the release workflow over in https://github.com/crystal-lang/distribution-scripts (see https://github.com/crystal-lang/distribution-scripts/blob/master/processes/crystal-release.md)

Reply:
The addition of a new branch (like when we release 1.3) needs to be triggered somehow. At that point we can also update the versions and website configuration.

Deletion is probably not even worth automating. That should never happen and if it does, it's because of manual interference.

Reply:
Thinking about this again, I think we should have all the configuration-level changes manually triggered (maybe it can be automatically triggered as part of the release process at some point, but that's not super important). 

The workflow could go like this:
1. Checkout https://github.com/crystal-lang/crystal-book
2. Create a new version branch from master and push it
3. Wait for deployment
4. Update latest redirects (S3 bucket website configuration) and `versions.json`

https://github.com/crystal-lang/distribution-scripts/ seems like a good place for that because we have everything else already collected there.

Reply:
I don't see much advantage to making it manually triggered (The creation of a branch in this repo is already a manual action BTW), but doesn't matter much to me. **I agree with your idea.**

This part is not critical for the moment, but clearly the way to proceed is to make a PR on distribution-scripts with these same scripts but slightly adapted. I am personally not interested in contributing to *distribution-scripts* but let me know if you think it's best that I do that.

And out of this PR only the change to *.github/workflows/deploy-book.yml* needs to be kept. But that should be applied *after* the other side (*distribution-scripts*) is finalized.